theatre/studio/src/panels/SequenceEditorPanel/DopeSheet/Right/BasicKeyframedTrack/KeyframeEditor/CurveEditorPopover/CurveEditorPopover.tsx

```diff
@@ -1,30 +1,116 @@
 import type {$IntentionalAny} from '@theatre/shared/utils/types'
+import fuzzySort from 'fuzzysort'
 
-const greaterThanZero = (v: number) => isFinite(v) && v > 0
+const presets = [
+  {label: 'Linear', value: '0.5, 0.5, 0.5, 0.5'},
+  {label: 'Back In Out', value: '0.680, -0.550, 0.265, 1.550'},
+  {label: 'Back In', value: '0.600, -0.280, 0.735, 0.045'},
+  {label: 'Back Out', value: '0.175, 0.885, 0.320, 1.275'},
+  {label: 'Circ In Out', value: '0.785, 0.135, 0.150, 0.860'},
+  {label: 'Circ In', value: '0.600, 0.040, 0.980, 0.335'},
+  {label: 'Circ Out', value: '0.075, 0.820, 0.165, 1'},
+  {label: 'Cubic In Out', value: '0.645, 0.045, 0.355, 1'},
+  {label: 'Cubic In', value: '0.550, 0.055, 0.675, 0.190'},
+  {label: 'Cubic Out', value: '0.215, 0.610, 0.355, 1'},
+  {label: 'Ease Out In', value: '.42, 0, .58, 1'},
+  {label: 'Expo In Out', value: '1, 0, 0, 1'},
+  {label: 'Expo Out', value: '0.190, 1, 0.220, 1'},
+  {label: 'Quad In Out', value: '0.455, 0.030, 0.515, 0.955'},
+  {label: 'Quad In', value: '0.550, 0.085, 0.680, 0.530'},
+  {label: 'Quad Out', value: '0.250, 0.460, 0.450, 0.940'},
+  {label: 'Quart In Out', value: '0.770, 0, 0.175, 1'},
+  {label: 'Quart In', value: '0.895, 0.030, 0.685, 0.220'},
+  {label: 'Quart Out', value: '0.165, 0.840, 0.440, 1'},
+  {label: 'Quint In Out', value: '0.860, 0, 0.070, 1'},
+  {label: 'Quint In', value: '0.755, 0.050, 0.855, 0.060'},
+  {label: 'Quint Out', value: '0.230, 1, 0.320, 1'},
+  {label: 'Sine In Out', value: '0.445, 0.050, 0.550, 0.950'},
+  {label: 'Sine In', value: '0.470, 0, 0.745, 0.715'},
+  {label: 'Sine Out', value: '0.390, 0.575, 0.565, 1'},
+]
 
 const Container = styled.div`
   display: flex;
+  flex-direction: column;
+  align-items: stretch;
+  width: 230px;
+`
+
+const InputContainer = styled.div`
+  display: flex;
+  gap: 8px;
+  align-items: center;
+`
+
+const OptionsContainer = styled.div`
+  display: grid;
+  grid-template-columns: 1fr 1fr;
   gap: 8px;
-  padding: 4px 8px;
-  height: 28px;
+  padding: 8px;
+  overflow: auto;
+  max-height: 130px;
+`
+
+const EasingOption = styled.div`
+  display: flex;
+  flex-direction: row;
   align-items: center;
+  gap: 4px;
+  overflow: hidden;
+
+  background: rgba(255, 255, 255, 0.1);
+  color: rgba(255, 255, 255, 0.75);
+  border-radius: 4px;
+
+  // The candidate preset is going to be applied when enter is pressed
+
+  &:focus {
+    box-shadow: 0 0 0 2px rgba(255, 255, 255, 0.4);
+  }
+
+  &:hover {
+    background: rgba(255, 255, 255, 0.2);
+  }
+
+  b {
+    text-decoration: underline;
+    // Default underline is too close to the text to be subtle
+    text-underline-offset: 2px;
+    text-decoration-color: rgba(255, 255, 255, 0.3);
+  }
 `
 
-const Label = styled.div`
-  ${propNameText};
-  white-space: nowrap;
+const EasingCurveContainer = styled.div`
+  display: flex;
+  padding: 6px;
+  background: rgba(255, 255, 255, 0.1);
+`
+
+const SearchBox = styled.input.attrs({type: 'text'})`
```

Comment:
I'm getting this on FF 94:

<img width="279" alt="image" src="https://user-images.githubusercontent.com/593118/145195268-4839d247-abf9-4c58-ab3e-d456caa57a7a.png">

Reply:
Firefox doesn't let grids overflow their own element when the height is fixed so we need an extra inner div for the grid.